api/client/src/main/java/jakarta/websocket/OnMessage.java

```diff
@@ -115,12 +115,35 @@
 
     /**
      * Specifies the maximum size of message in bytes that the method this annotates will be able to process, or -1 to
-     * indicate that there is no maximum. The default is -1. This attribute only applies when the annotation is used to
-     * process whole messages, not to those methods that process messages in parts or use a stream or reader parameter
-     * to handle the incoming message. If the incoming whole message exceeds this limit, then the implementation
-     * generates an error and closes the connection using the reason that the message was too big.
+     * indicate that no maximum has been configured. The default is -1. This attribute only applies when the annotation
+     * is used to process whole messages, not to those methods that process messages in parts or use a stream or reader
+     * parameter to handle the incoming message. If the incoming whole message exceeds this limit, then the
+     * implementation generates an error and closes the connection using the reason that the message was too big.
+     * <p>
+     * This attribute is ignored unless {@link #maxMessageBufferSize()} returns {@code -1}.
+     * <p>
+     * This attribute should not be set to a value larger than {@code Integer#MAX_VALUE}.
      *
      * @return the maximum size in bytes.
+     *
+     * @deprecated Use {@link #maxMessageBufferSize()}
      */
+    @Deprecated
     public long maxMessageSize() default -1;
```

Comment:
Wouldn't it be easier to just make the default `Integer.MAX_VALUE` instead of `-1`?

Reply:
I've re-worked the solution along the lines of your option 2.